pkg/scheduler/plugins/resource-strategy-fit/resource_strategy_fit_test.go

```diff
@@ -0,0 +1,665 @@
+
+	n5 := util.BuildNode("n5", api.BuildResourceList("4", "4Gi", []api.ScalarResource{{Name: "pods", Value: "10"}}...), make(map[string]string))
+	addResource(n5.Status.Allocatable, GPU, "10")
+	addResource(n5.Status.Allocatable, FOO, "5")
+	n6 := util.BuildNode("n6", api.BuildResourceList("4", "4Gi", []api.ScalarResource{{Name: "pods", Value: "10"}}...), make(map[string]string))
+	addResource(n6.Status.Allocatable, FOO, "5")
+	addResource(n6.Status.Allocatable, FOO, "10")
+
+	pg1 := util.BuildPodGroup("pg1", "c1", "c1", 0, nil, "")
+	queue1 := util.BuildQueue("c1", 1, nil)
+
+	tests := []struct {
+		uthelper.TestCommonStruct
+		arguments framework.Arguments
+		expected  map[string]map[string]float64
+	}{
+		{
+			TestCommonStruct: uthelper.TestCommonStruct{
+
+				Name:      "single job",
+				Plugins:   map[string]framework.PluginBuilder{PluginName: New},
+				PodGroups: []*schedulingv1.PodGroup{pg1},
+				Queues:    []*schedulingv1.Queue{queue1},
+				Pods:      []*v1.Pod{p1, p2, p3, p4},
+				Nodes:     []*v1.Node{n1, n2, n3, n4},
+			},
+			arguments: framework.Arguments{
+				"ResourceStrategyFitPlusWeight": 10,
+				"resources": map[string]interface{}{
+					"nvidia.com/gpu": map[string]interface{}{
+						"type":   "MostAllocated",
+						"weight": 1,
+					},
+					"example.com/foo": map[string]interface{}{
+						"type":   "LeastAllocated",
+						"weight": 1,
+					},
+				},
+			},
+			expected: map[string]map[string]float64{
+				"c1/p1": {
+					"n1": 0,
+					"n2": 0,
+					"n3": 800,
+					"n4": 600,
+				},
+				"c1/p2": {
+					"n1": 0,
+					"n2": 0,
+					"n3": 700,
+					"n4": 400,
+				},
+				"c1/p3": {
+					"n1": 200,
+					"n2": 400,
+					"n3": 0,
+					"n4": 0,
+				},
+				"c1/p4": {
+					"n1": 300,
+					"n2": 600,
+					"n3": 0,
+					"n4": 0,
+				},
+			},
+		},
+		{
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:      "single job",
+				Plugins:   map[string]framework.PluginBuilder{PluginName: New},
+				PodGroups: []*schedulingv1.PodGroup{pg1},
+				Queues:    []*schedulingv1.Queue{queue1},
+				Pods:      []*v1.Pod{p5},
+				Nodes:     []*v1.Node{n5, n6},
+			},
+			arguments: framework.Arguments{
+				"ResourceStrategyFitPlusWeight": 10,
+				"resources": map[string]interface{}{
+					"nvidia.com/gpu": map[string]interface{}{
+						"type":   "MostAllocated",
+						"weight": 1,
+					},
+					"example.com/foo": map[string]interface{}{
+						"type":   "LeastAllocated",
+						"weight": 2,
+					},
+				},
+			},
+			expected: map[string]map[string]float64{
+				"c1/p5": {
+					"n5": 266.66666666,
+					"n6": 399.99999999,
+				},
+			},
+		},
+	}
+
+	trueValue := true
+	for i, test := range tests {
+		tiers := []conf.Tier{
+			{
+				Plugins: []conf.PluginOption{
+					{
+						Name:             PluginName,
+						EnabledNodeOrder: &trueValue,
+						Arguments:        test.arguments,
+					},
+				},
+			},
+		}
+		ssn := test.RegisterSession(tiers, nil)
+		for _, job := range ssn.Jobs {
```

Comment:
You should write this test like
```
	for i, test := range tests {
		t.Run(test.Name, func(t *testing.T) {
			test.Plugins = plugins
			test.RegisterSession(tiers, nil)
			defer test.Close()
			test.Run([]framework.Action{New()})
			if err := test.CheckAll(i); err != nil {
				t.Fatal(err)
			}
		})
	}
```
https://github.com/volcano-sh/volcano/blob/068b7bc3d8ecfc8274a82e4f458bbb84fac883bf/pkg/scheduler/actions/allocate/allocate_test.go#L259-L269
Not only NodeOrder

Reply:
done